src/components/global/upload_tabs/azure/AuthStepBasic.vue

```diff
@@ -0,0 +1,265 @@
+<template>
+  <v-stepper-content step="1">
+    <br />
+    <v-form>
+      <v-select
+        :items="auth_methods"
+        v-model="auth_method_model"
+        @change="change_auth_method"
+        return-object
+        item-text="text"
+        item-value="value"
+        label="Select Auth Method"
+        dense
+      ></v-select>
+    </v-form>
+    <v-form v-model="valid">
+      <v-text-field
+        :value="container_name"
+        @input="change_container_name"
+        label="Container Name (optional)"
+        lazy-validation="lazy"
+        :rules="[]"
+      />
+    </v-form>
+    <v-form
+      v-model="valid"
+      v-if="auth_method_model && auth_method_model.value == 'sas'"
+    >
+      <v-text-field
+        :value="account_name"
+        @input="change_account_name"
+        label="Storage Account Name"
+        lazy-validation="lazy"
+        :rules="[req_rule]"
+      />
+      <v-text-field
+        :value="account_suffix"
+        @input="change_account_suffix"
+        label="Account Suffix"
+        lazy-validation="lazy"
+        :rules="[req_rule]"
+      />
+      <v-text-field
+        :value="shared_access_signature"
+        @input="change_shared_access_signature"
+        label="Shared Access Signature"
+        :rules="[req_rule]"
+        :append-icon="show_secret ? 'mdi-eye' : 'mdi-eye-off'"
+        :type="show_secret ? 'text' : 'password'"
+        @click:append="show_secret = !show_secret"
+      />
+      <v-text-field
+        :value="get_full_account_url()"
+        label="Full Storage Account URL"
+        readonly
+        outlined
+      />
+    </v-form>
+    <v-form
+      v-model="valid_conn_string"
+      v-if="auth_method_model && auth_method_model.value == 'conn_string'"
+    >
+      <v-text-field
+        :value="connection_string"
+        @input="change_connection_string"
+        label="Connection String"
+        lazy-validation="lazy"
+        :rules="[req_rule]"
+      />
+    </v-form>
+    <v-btn
+      color="primary"
+      :disabled="!(valid || valid_conn_string)"
+      @click="$emit('auth-basic')"
+      class="my-2 mr-3"
+    >
+      Login
+    </v-btn>
+  </v-stepper-content>
+</template>
+
+<script lang="ts">
+import Vue from "vue";
+import Component from "vue-class-component";
+import { LocalStorageVal } from "../../../../utilities/helper_util";
+import { get_blob_account_url } from "../../../../utilities/azure_util";
+import { Dictionary } from "vue-router/types/router";
+
+type MultipleSelectObject = {
+  value: string;
+  text: string;
+};
+
+// We declare the props separately to make props types inferable.
+const Props = Vue.extend({
+  props: {
+    auth_method: Object,
+    account_name: String,
+    connection_string: String,
+    container_name: String,
+    account_suffix: String,
+    shared_access_signature: String
+  }
+});
+
+/** Localstorage keys */
+const local_auth_method = new LocalStorageVal<MultipleSelectObject>(
```

Comment:
Instead of creating an object here and then strings for the rest of the account information, could you create an `azure_auth` localStorageVal and just store all these on that object?